### PR TITLE
#265 - activity tags

### LINKED ISSUE #265: Activities tags and assigned people
Activities tags should be managable form admin panel.
We should add additionally something like people which are connected with activity.

### PR DESCRIPTION
**Pull request by kamilpiech97**
Should close #265.
Added tags for tagging our activities. We can create tag for tagging people (`as_person` flag). In tag we can set visibility on main page of activities (`is_primary` flag) also.
Should be opened after #254.

### CODE REVIEW

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Text "Tags" is not translated

image

2. Can we add sorting to tags by name?

3. Please translate this

image

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. Create a tag
2. Add the tag to the activity
3. Delete the tag from tag index
4. Go to activity edit view where the tag was attached
5. Tag list is empty

**Comment by kamilpiech97** [comment]
> 1. Create a tag
> 2. Add the tag to the activity
> 3. Delete the tag from tag index
> 4. Go to activity edit view where the tag was attached
> 5. Tag list is empty

Fixed

**Review by EwelinaSkrzypacz — APPROVED** [review]

File: app/Models/CaseStudy.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
  * @property array $name
  * @property array $description
  * @property string $photo
- * @property string $slug
+ * @property array $slug
```
Comment (AleksandraKozubal) [inline app/Models/CaseStudy.php:15]: What is the reason to change it to array? we're not translating slugs

File: database/factories/CaseStudyFactory.php [hunk 2]
```diff
@@ -22,7 +22,7 @@ public function definition(): array
             "photo" => sprintf("%s/%s", "factory", "case_study.jpg"),
             "published" => $this->faker->boolean,
             "company" => $this->faker->company,
-            "slug" => $this->faker->slug,
+            "slug" => $this->translations($locales, $this->faker->slug),
```
Comment (AleksandraKozubal) [inline database/factories/CaseStudyFactory.php:25]: Same here

File: database/migrations/2024_06_25_101545_change_slug_in_case_studies_table.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?php
```
Comment (AleksandraKozubal) [inline database/migrations/2024_06_25_101545_change_slug_in_case_studies_table.php:1]: Are these remnants from another branch?

File: app/Models/CaseStudy.php (hunk 1 above)
Comment (kamilpiech97) [reply app/Models/CaseStudy.php]: fixed, came from previous pull request (before your changes in code)

File: database/factories/CaseStudyFactory.php (hunk 2 above)
Comment (kamilpiech97) [reply database/factories/CaseStudyFactory.php]: fixed, came from previous pull request (before your changes in code)

File: database/migrations/2024_06_25_101545_change_slug_in_case_studies_table.php (hunk 3 above)
Comment (kamilpiech97) [reply database/migrations/2024_06_25_101545_change_slug_in_case_studies_table.php]: fixed, came from previous pull request (before your changes in code)

**Review by AleksandraKozubal — DISMISSED** [review]

**Review by AleksandraKozubal — APPROVED** [review]

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Pull request merged**